doctor_visits/delphi_doctor_visits/sensor.py

```diff
@@ -61,13 +61,15 @@ def fill_dates(y_data, dates):
         cols = y_data.columns
 
         if first_date not in y_data.index:
-            y_data = y_data.append(
+            y_data = pd.concat([
+                y_data,
                 pd.DataFrame(dict.fromkeys(cols, 0.0), columns=cols, index=[first_date])
-            )
+            ])
         if last_date not in y_data.index:
-            y_data = y_data.append(
+            y_data = pd.concat([
+                y_data,
                 pd.DataFrame(dict.fromkeys(cols, 0.0), columns=cols, index=[last_date])
-            )
+            ])
```

Comment:
question: how big is `y_data` typically?

`pd.concat` has a bunch of expensive overhead that increases with the number of rows, so if `y_data` is big it might be worth building up a list of data frames and then concatenating just once at the end.

Reply:
Whoops, thanks for pointing that out!